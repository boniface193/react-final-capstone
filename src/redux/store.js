import { configureStore } from '@reduxjs/toolkit';
import cryptoSlice from './balance/balanceSlice';
import detailSlice from './balance/detailSlice';

const store = configureStore({
  reducer: {
    cryptos: cryptoSlice,
    cryptoDetail: detailSlice,
  },
});

export default store;
