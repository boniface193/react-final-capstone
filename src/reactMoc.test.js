import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './redux/store';
import Category from './components/Categories';

describe('Category component', () => {
  it('matches the snapshot', () => {
    const cryptos = [
      {
        symbol: 'BTC',

      },
      {
        symbol: 'ETH',
      },
    ];

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Category cryptos={cryptos} />
        </Router>
      </Provider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
