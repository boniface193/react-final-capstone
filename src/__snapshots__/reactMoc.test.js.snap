// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component matches the snapshot 1`] = `
<section
  className="text-white card"
>
  <article>
    <a
      href="/BTC"
      onClick={[Function]}
    >
      <i
        className="fa-regular fa-circle-right cursor-pointer"
      />
    </a>
    <div
      className="card-body d-flex"
    >
      <aside>
        <h6>
          ExchangeShortName
        </h6>
        <p />
      </aside>
      <aside
        className="pl-2"
      >
        <h3>
          symbol
        </h3>
        <p
          className="text-yellow"
        >
          BTC
        </p>
      </aside>
    </div>
  </article>
  <article>
    <a
      href="/ETH"
      onClick={[Function]}
    >
      <i
        className="fa-regular fa-circle-right cursor-pointer"
      />
    </a>
    <div
      className="card-body d-flex"
    >
      <aside>
        <h6>
          ExchangeShortName
        </h6>
        <p />
      </aside>
      <aside
        className="pl-2"
      >
        <h3>
          symbol
        </h3>
        <p
          className="text-yellow"
        >
          ETH
        </p>
      </aside>
    </div>
  </article>
</section>
`;
